developer_guides/dsp_core/index.rst

```diff
@@ -0,0 +1,81 @@
+.. _dsp_core:
```

Comment:
@keyonjie thanks for starting this document but I think we should focus more on how the cores should be managed instead of how the users (ie widgets/pipelines etc) use them.

So I would make the document focus first on how the cores should be managed ie with ref_counting to account for the fact that core must be powered on until the last user releases the ref_count. And then follow up it with the example of how topology components use ref_counting.

Reply:
> @keyonjie thanks for starting this document but I think we should focus more on how the cores should be managed instead of how the users (ie widgets/pipelines etc) use them.
> 
> So I would make the document focus first on how the cores should be managed ie with ref_counting to account for the fact that core must be powered on until the last user releases the ref_count. And then follow up it with the example of how topology components use ref_counting.

That's true, but some context about how and why a widget/component should take a ref_counting of the cores could help to understand the full picture, IMHO.